tests/testparamdepends.py

```diff
@@ -820,6 +820,27 @@ def method1(self):
         assert method1_count == 0
         assert method2_count == 1
 
+    def test_param_depends_class_with_len(self):
+        # https://github.com/holoviz/param/issues/747
+
+        count = 0
+
+        class P(param.Parameterized):
+            x = param.Parameter()
+
+            @param.depends('x', watch=True)
+            def debug(self):
+                nonlocal count
+                count += 1
+
+            # bool(P()) evaluates to False
+            def __len__(self):
+                return 0
+
+        p = P()
+        p.x = 1
+        assert count == 1
```

Comment:
I would rewrite this to something like this:
``` python
    import param

    class P(param.Parameterized):
        x = param.Parameter()
        count = param.Number(0)

        @param.depends('x', watch=True)
        def debug(self):
            self.count += 1

        # bool(P()) evaluates to False
        def __len__(self):
            return 0

    p = P()
    p.x = 1
    assert p.count == 1
```

Reply:
No distinct advantages.

I think this aligns more with a real-world example and as such it is easier (at least for me) to read and understand.

Reply:
I think I still prefer the approach that separates concerns more.

>  it is easier (at least for me) to read and understand.

Oh I'm 100% sure you have no problem understanding that test! :)

Reply:
It just looks like you have gone out of your way to do it this way over the parametrized class approach. This could lead to confusion in the future.  

It is very nit, and as I said it was something I would have done.